File: java/ecbenchmark/src/ecbenchmark/cec10/F03.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package ecbenchmark.cec10;

import ecbenchmark.function.Ackley;
import ecbenchmark.Function;
import ecbenchmark.operator.Shifted;

/**
 *
 * @author jcrada
 */
public class F03 extends CecFunction {

    private Function ackley;

    public F03(int dimensions) {
        super("F03", "Shifted Ackley's Function", dimensions,
                -32, 32, true, new CecRandom(3l), 0);
        double[] shift = CecMath.ShiftVector(dimensions, getMinimum(), getMaximum(), getRandomizer());
        ackley = new Shifted(shift, 0, new Ackley());
    }

    public double f(final double[] x) {
        return ackley.f(x);
    }
}

File: cpp/ecbenchmark/src/cec10/F03.cpp
#include "ecbenchmark/cec10/F03.h"
#include "ecbenchmark/cec10/CecMath.h"
#include "ecbenchmark/operator/Shifted.h"
#include "ecbenchmark/function/Ackley.h"

namespace ecb {
    namespace cec10 {

        F03::F03(int dimensions)
        : CecFunction("F03", "Shifted Ackley's Function", dimensions,
        -32, 32, true, 0, new CecRandom(3l)) {
            std::vector<scalar> shift = CecMath::ShiftVector(dimensions, minimumDomain(), maximumDomain(), randomizer());
            ackley = new Shifted(shift, 0, new Ackley());
        }

        F03::~F03() {
            delete ackley;
        }

        scalar F03::f(const std::vector<scalar>& x) {
            return ackley->f(x);
        }
    }
}

